libvast/src/system/spawn_disk_monitor.cpp

```diff
@@ -0,0 +1,63 @@
+#include "vast/system/spawn_disk_monitor.hpp"
+
+#include "vast/concept/parseable/vast/si.hpp"
+#include "vast/defaults.hpp"
+#include "vast/logger.hpp"
+#include "vast/path.hpp"
+#include "vast/system/disk_monitor.hpp"
+#include "vast/system/node.hpp"
+#include "vast/system/node_control.hpp"
+#include "vast/system/spawn_arguments.hpp"
+
+#include <caf/actor.hpp>
+#include <caf/event_based_actor.hpp>
+#include <caf/expected.hpp>
+#include <caf/scoped_actor.hpp>
+#include <caf/settings.hpp>
+#include <caf/typed_event_based_actor.hpp>
+
+#include "caf/error.hpp"
+
+namespace vast::system {
+
+maybe_actor
+spawn_disk_monitor(system::node_actor* self, spawn_arguments& args) {
+  VAST_TRACE(VAST_ARG(args));
+  auto [index, archive]
+    = self->state.registry.find_by_label("index", "archive");
+  if (!index)
+    return make_error(ec::missing_component, "index");
+  if (!archive)
+    return make_error(ec::missing_component, "archive");
+  auto hiwater_str
+    = caf::get_or(args.inv.options, "vast.start.disk-budget-high", "0KiB");
+  auto lowater_str
+    = caf::get_or(args.inv.options, "vast.start.disk-budget-low", "0KiB");
+  uint64_t hiwater, lowater;
+  if (parsers::bytesize(hiwater_str, hiwater))
+    return make_error(ec::invalid_argument,
+                      "could not parse disk budget: " + hiwater_str);
+  if (parsers::bytesize(lowater_str, lowater))
+    return make_error(ec::invalid_argument,
+                      "could not parse disk budget: " + lowater_str);
```

Comment:
Shouldn't these `if` statements be inverted? The parsers return `true` on success.

Reply:
Empirically they don't, unless I've messed something up while doing the PR cleanup. I'll double-check to be sure.

Reply:
Yeah, please double-check. If you look at the framework, all parsers return a `bool` return value; `true` means parsing success and `false` failure.